examples/async_tasks/async_web_request.rs

```diff
@@ -0,0 +1,143 @@
+//! This example shows how to use the ECS and the [`AsyncComputeTaskPool`]
+//! to spawn, poll, and complete web request.
+
+use bevy::{
+    prelude::*,
+    tasks::{block_on, futures_lite::future, AsyncComputeTaskPool, Task},
+};
+use ehttp::{Request, Response};
+
+fn main() {
+    App::new()
+        .add_plugins(DefaultPlugins)
+        .init_resource::<ApiTimer>()
+        .add_systems(Startup, setup_env)
+        .add_systems(
+            Update,
+            (send_request, handle_request, handle_response, update_text),
+        )
+        .run();
+}
+
+/// wrap for ehttp request
+#[derive(Component, Debug, Clone, Deref, DerefMut)]
+pub struct HttpRequest(pub Request);
+
+/// wrap for ehttp response
+#[derive(Component, Debug, Clone, Deref, DerefMut)]
+pub struct HttpResponse(pub Response);
+
+/// wrap for ehttp error
+#[derive(Component, Debug, Clone, Deref, DerefMut)]
+pub struct HttpResponseError(pub String);
+
+/// task for ehttp response result
+#[derive(Component)]
+pub struct RequestTask(pub Task<Result<Response, ehttp::Error>>);
+
+/// timer component
+#[derive(Resource, Deref, DerefMut)]
+pub struct ApiTimer(pub Timer);
+
+impl Default for ApiTimer {
+    fn default() -> Self {
+        Self(Timer::from_seconds(3.0, TimerMode::Repeating))
+    }
+}
+
+/// system that will tick timer and send request each time timer has just finished
+fn send_request(mut commands: Commands, time: Res<Time>, mut timer: ResMut<ApiTimer>) {
+    timer.tick(time.delta());
+
+    if timer.just_finished() {
+        let req = ehttp::Request::get("https://api.ipify.org?format=json");
+        commands.spawn(HttpRequest(req));
+    }
+}
+
+/// system that will spawn task for each request
+fn handle_request(
+    mut commands: Commands,
+    requests: Query<(Entity, &HttpRequest), Without<RequestTask>>,
+) {
+    let thread_pool = AsyncComputeTaskPool::get();
```

Comment:
this is not a thread pool. and I think `AsyncComputeTaskPool` would be the wrong task pool to use for this kind of task